AnnoMapEditor/MapTemplates/IslandType.cs

```
@@ -3,70 +3,50 @@
 
-        public bool IsNormalOrStarter => value == nameof(Starter) || value == nameof(Normal);
-        public bool IsSameWithoutOil(IslandType that) => IsNormalOrStarter && that.IsNormalOrStarter || value == that.value;
+        public readonly string Name;
 
-        public IslandType(string? type)
-        {
-            ElementValue = (short?)RandomIDs.IndexOf((x) => x == type) ?? 0;
-            if (type == "Starter" || type == "ThirdParty" || type == "Decoration")
-            {
-                value = type;
-                return;
-            }
-            if (type == "PirateIsland" || type == "Pirate")
-            {
-                value = "PirateIsland";
-                return;
-            }
-            if (type == "Cliff")
-            {
-                value = "Cliff";
-                return;
-            }
+        public readonly short ElementValue;
+
+        public bool IsNormalOrStarter => this == Starter || this == Normal;
 
-            value = "Normal";
+        public bool IsSameWithoutOil(IslandType that) => IsNormalOrStarter && that.IsNormalOrStarter;
+
+
+        private IslandType(string name, short elementValue)
+        {
+            ElementValue = elementValue;
+            Name = name;
         }
 
-        public IslandType(short? type)
+        public static IslandType FromElementValue(short elementValue)
         {
-            ElementValue = type ?? 0;
-            if (type is null)
+            return elementValue switch
             {
-                value = "Normal";
-                return;
-            }
-
-            value = RandomIDs[Math.Clamp((int)type, 0, RandomIDs.Length - 1)];
+                0 => Normal,
+                1 => Starter,
+                2 => Decoration,
+                3 => ThirdParty,
+                4 => PirateIsland,
+                5 => Cliff,
+                _ => throw new ArgumentException($"{elementValue} is not a valid element value for IslandType.")
```

It's better to not use exceptions for input data, otherwise manually touched XMLs may not open properly anymore.

---

I am torn on this one. If someone modifies an XML and enters an invalid value, we should not silently ignore it. Defaulting to some other IslandType might not be in the users interest.

---

Write a log, possibly show an error. That's what a pro user would want.

Everyone else is simply happy to open stuff and see what's in.
Most information is immediately visible in the editor, so silent changes are not problematic IMO.

By crashing the application, you only help those who are more capable.

---

I changed the behaviour for `IslandType`. Does the same criticism apply to `IslandSize` too? It feels more critical to me.

---

Yes, that's a general point we should keep in mind.
Especially since there are some handwritten mods out there already with slightly broken data ^^

---

Done